pkg/api/controller.go

```diff
@@ -215,9 +217,81 @@ func (c *Controller) Run(reloadCtx context.Context) error {
 	return server.Serve(listener)
 }
 
+// Will create a cache database using the configuration file.
+// Pass a non-nil first argument to override the config (for testing purposes, mostly).
+func (c *Controller) CreateCacheDatabaseDriver(configOverride interface{}, log log.Logger) (database.Driver, error) {
```

Comment:
Why use panics in this functions, if it potentially returns errors?

Reply:
Changed logic to return errors instead